packages/@react-spectrum/textfield/src/TextArea.tsx

```diff
@@ -26,31 +27,80 @@ function TextArea(props: SpectrumTextFieldProps, ref: RefObject<TextFieldRef>) {
     isReadOnly = false,
     isRequired = false,
     onChange,
+    labelPosition,
     ...otherProps
   } = props;
 
+  let fallbackRef = useRef<TextFieldRef>(null);
+  if (!ref) {
+    ref = fallbackRef;
+  }
+
   // not in stately because this is so we know when to re-measure, which is a spectrum design
   let [inputValue, setInputValue] = useControlledState(props.value, props.defaultValue, () => {});
 
   let inputRef = useRef<HTMLTextAreaElement>();
 
+  // Store styleProps to apply styles appropriately on height change.
+  let {styleProps} = useStyleProps(props);
+  let initialHeightsRef = useRef(null);
+
   let onHeightChange = useCallback(() => {
+    let input = inputRef.current;
+    let field = ref.current.UNSAFE_getDOMNode();
+    if (initialHeightsRef.current === null) {
+      initialHeightsRef.current = {
+        inputHeight: input.offsetHeight,
+        fieldHeight: field.offsetHeight,
+        diff: field.offsetHeight - input.offsetHeight
+      };
+    }
+    let {inputHeight, fieldHeight, diff} = initialHeightsRef.current;
+
+    // label or helptext height might change so we have to recalculate
+    let labelHeight = labelPosition !== 'side' && !field.firstElementChild.contains(input) ? (field.firstElementChild as HTMLElement).offsetHeight : 0;
+    let helpTextHeight = !field.lastElementChild.contains(input) ? (field.lastElementChild as HTMLElement).offsetHeight : 0;
+    
     if (isQuiet) {
-      let input = inputRef.current;
       let prevAlignment = input.style.alignSelf;
       input.style.alignSelf = 'start';
       input.style.height = 'auto';
-      input.style.height = `${input.scrollHeight}px`;
+      input.style.height = `${Math.max(input.scrollHeight, inputHeight)}px`;
       input.style.alignSelf = prevAlignment;
+      for (const [key, value] of Object.entries(styleProps.style)) {
+        switch (key) {
+          case 'height':
+          case 'maxHeight':
+            field.style[key] = input.offsetHeight + labelHeight + helpTextHeight <= fieldHeight ? value : '';
+            break;
+        }
+      }
+    } else {  
+      for (const [key, value] of Object.entries(styleProps.style)) {
+        switch (key) {
+          case 'height':
+            field.style[key] = input.offsetHeight + labelHeight + helpTextHeight <= fieldHeight ? value : `calc(${value} - ${fieldHeight - (Math.max(inputHeight, input.offsetHeight) + labelHeight + helpTextHeight)}px)`;
```

Comment:
noticed that the helptext doesn't remain directly under the textarea if the textarea is reduced below its initial set height, digging to see if the calculation here can be modified to accommodate that

Reply:
We may have to override `field.style.height = ‘’`